### PR TITLE
Properly handle serialization deserizliation for redis transaction api.

### PR DESCRIPTION
**Pull request by mukundansundar**
# Description

Found an issue for Transaction API via Java Integration Tests. 

When the state is stored via an **upsert** call made and the got back via the normal getState API call, it can be observed that there is an extra Json Marshal happening in the code. 

Example:
```json
data: "\"{\\\"propertyA\\\":\\\"data in property A\\\",\\\"propertyB\\\":\\\"data in property B\\\"}\""
```
Instead of 
```json
data: "{\"propertyA\":\"data in property A\",\"propertyB\":\"data in property B\"}"
````
The second output is what we have when we have a normal save and get state call done via Dapr. 

The code change here, checks if the incoming request value is a byte array or not and then properly serializes the value. 
## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: state/redis/redis.go [hunk 1]
```diff
@@ -298,8 +298,16 @@ func (r *StateStore) Multi(request *state.TransactionalStateRequest) error {
 	for _, o := range request.Operations {
 		if o.Operation == state.Upsert {
 			req := o.Request.(state.SetRequest)
-			b, _ := r.json.Marshal(req.Value)
-			pipe.Set(req.Key, b, defaultExpirationTime)
+
```
Comment (mukundansundar) [inline state/redis/redis.go:301]: changes similar to https://github.com/dapr/components-contrib/pull/460/files#diff-ae10e12c8f4b65b24b405cfe5a7ed386R255

File: state/redis/redis.go [hunk 2]
```diff
@@ -298,8 +298,16 @@ func (r *StateStore) Multi(request *state.TransactionalStateRequest) error {
...
+			var bt []byte
+			b, ok := req.Value.([]byte)
+			if ok {
+				bt = b
+			} else {
+				bt, _ = r.json.Marshal(req.Value)
+			}
```
Comment (tcnghia) [inline state/redis/redis.go:308]:
```suggestion
			bt, ok := req.Value.([]byte)
			if !ok {
				bt, _ = r.json.Marshal(req.Value)
			}
```

File: state/azure/cosmosdb/cosmosdb.go [hunk 3]
```diff
@@ -306,6 +306,7 @@ func (c *StateStore) Multi(request *state.TransactionalStateRequest) error {
 		if o.Operation == state.Upsert {
 			req := o.Request.(state.SetRequest)
 
+			// Value need not be marshaled here. It is handled by cosmosdb client.
```
Comment (mukundansundar) [inline state/azure/cosmosdb/cosmosdb.go:309]: https://github.com/a8m/documentdb/blob/59a8c720ee4a28e9aed5fabde2949a85bb3a3238/client.go#L166

File: state/postgresql/postgresdbaccess.go [hunk 4]
```diff
@@ -93,14 +93,15 @@ func (p *postgresDBAccess) setValue(req *state.SetRequest) error {
 		return fmt.Errorf("missing key in set operation")
 	}
 
-	var valueBytes []byte
-
 	// Convert to json string
-	valueBytes, err = json.Marshal(req.Value)
-	if err != nil {
-		return err
+	var bt []byte
```
Comment (youngbupark) [inline state/postgresql/postgresdbaccess.go:97]: if this code is used in multiple places, it would be great to create shared util.

**Review by youngbupark — CHANGES_REQUESTED** [review]

**Review by youngbupark — CHANGES_REQUESTED** [review]

File: state/utils/utils.go [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package utils
+
+func Marshal(val interface{}, marshaler func(interface{}) ([]byte, error)) ([]byte, error) {
+	var err error
+	bt, ok := val.([]byte)
+	err = nil
```
Comment (youngbupark) [inline state/utils/utils.go:11]: no need to set error to nil. in go, default error value is nil.

if you want to set explicitly, 

I would change line 9.
```
var err error = nil
```

**Review by youngbupark — CHANGES_REQUESTED** [review]

**Review by artursouza — APPROVED** [review]

**Review by youngbupark — APPROVED** [review]

**Pull request merged**